src/content/en/2020/javascript.md

```diff
@@ -297,7 +296,52 @@ Both are very sound guesses, and it seems the situation hasn't changed much on e
 
 ### Frameworks
+When we look at the frameworks, we also don't see much of a dramatic change in terms of adoption in the main frameworks that were highlighted last year. Vue.js has seen a significant increase, and AMP grew a bit, but most of them are more less where they were a year ago. 
 
+{# Compare same frameworks from last year's chapter to this year in bar chart? #}
+
+It's worth noting that the detection issue that was noted last year still applies, and still impacts th results here. It's possible that there _has_ been a significant change in popularity for a few more of these tools, but we just don't see it with the way the data is currently collected.
+
+### What it all means?
+More interesting to me than the popularity of the tools themselves is the impact they have on the things we build.
+
+First, it's worth noting that while we may think of the usage of one tool versus another, in reality, we rarely only use a single library or framework in production. Only 21% of pages analyzed report only one library or framework. Two or three frameworks are pretty common, and the long-tail gets very long, very quickly.
+
+{# Distribution of number of JS frameworks or libraries on a page #}
+
+When we look at the common combinations that we see in production, most of them are to be expected. Knowing jQuery's dominance, it's unsurprising that most of the popular combinations include jQuery and any number of jQuery related plugins.
+
+{# Top 20? Combos with percentage #}
+
+We do also see a fair amount of more "modern" frameworks, like React, Vue and Angular—paired with jQuery, like as a result of either migration or third-parties including one or the other.
+
+{# React/Angular/Vue % with and without jQuery? #}
+
+More importantly, all these tools typically mean more code and more processing time.
+
+Looking specfically at the frameworks in use, we see that the median JavaScript bytes for pages using them varies dramatically depending on _what_ is being used.
+
+The graph below shows the median bytes for pages where any of the top 35 most commonly detected frameworks were found, broken down by client.
+
+{# Median bytes by JS framework #}
+
+On one of the spectrum are frameworks like React or Angular or Ember, which tend to ship a lot of code regardless of the client. On the other end, we see minimalist frameworks like Alpine.js and Svelte showing very promising results. Defaults are very important, and it seems that by starting with highly performant defaults, Svelte and Alpine are both succeeding (so far...the sample size is pretty small) in creating a lighter set of pages.
+
+We get a very similar picture when looking at main thread time for pages where these tools were detected.
+
+{# Median main thread by JS framework #}
+
+Ember's mobile main thread time jumps out and kind of distorts the graph with how long it takes. Pulling it out makes the picture a bit easier to understand. 
```

Comment:
```suggestion
Ember's mobile main thread execution time jumps out and kind of distorts the graph with how long it takes. Pulling it out makes the picture a bit easier to understand. 
```

Reply:
I'm gonna leave that out here as it's not just execution time being recorded, but any main thread time (parse/compile/execution).